src/main/java/org/cru/cas/client/integration/EntityPreservingSingleSignOutFilter.java

```diff
@@ -0,0 +1,95 @@
+package org.cru.cas.client.integration;
+
+import static org.cru.cas.client.integration.Util.nullToEmpty;
+
+import java.io.IOException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.jasig.cas.client.session.SessionMappingStorage;
+import org.jasig.cas.client.session.SingleSignOutFilter;
+
+/**
+ * A filter that behaves the same as {@link SingleSignOutFilter} except that
+ * it does not consume the entity input stream unless the path matches
+ * the configured logout path.
+ *
+ * This assumes that the ticket is sent in a GET request, which might be wrong for SAML
+ * (I'm not sure),
+ * but is fine for the native CAS protocol.
+ *
+ * Works around https://github.com/apereo/java-cas-client/issues/210.
+ */
+public class EntityPreservingSingleSignOutFilter implements Filter {
```

Comment:
I don't see this filter referenced in the [web.xml](https://drive.google.com/drive/folders/1s9BcegxJRl6IfXUoRvMQRTMPTQGlFcM4). Should it be?

Reply:
I now see from the PR Description that, no, due to some yet to be identified issue with PSHR, single-logout (SLO) will not be made available in this PR.